test/cppsuite/test_harness/test.h

```diff
@@ -105,10 +104,22 @@ class test {
               TABLE_OPERATION_TRACKING, SCHEMA_TRACKING_TABLE_CONFIG, TABLE_SCHEMA_TRACKING);
             /* Make sure the tracking component is loaded first to track all activities. */
             _components.insert(_components.begin(), _workload_tracking);
-        } else
-            _workload_tracking = nullptr;
-        /* Tell the workload generator whether tracking is enabled. */
-        _workload_generator->set_tracker(_workload_tracking);
+            /* Tell the workload generator whether tracking is enabled. */
+            _workload_generator->set_tracker(_workload_tracking);
+        }
+
+        /* Create the timestamp manager if required. */
+        testutil_check(_configuration->get_bool(ENABLE_TIMESTAMP, enable_timestamp));
```

Comment:
I think this should be checked in the timestamp_manager rather than in the test. Then the workload_generator can call the timestamp_manager to know if its enabled rather than setting a flag on the workload_generator from the test.

Reply:
For the tracking component, we check that it is enabled outside of it too.
Only if they are marked as enabled in the config, we instantiate them.

Reply:
I will try to implement this bit differently, I will update the PR soon.

Reply:
Move this inside timestamp manager.

Reply:
Add timestamp manager to the components list directly. 👍